dcpy/test/metadata/test_metadata.py

```diff
@@ -59,31 +62,55 @@ def bbl(boro_code, block, lot):
 }
 
 
-def fake_row(columns: list[models.Column]):
+def _fake_row(columns: list[models.Column]):
     row = {}
 
     found_bbl_parts = {}
     bbl_parts = {"boro_code", "block", "lot"}
+    found_bbl_name = ""
     for c in columns:
         if c.data_type == "bbl":
-            pass
+            found_bbl_name = c.name
+        elif c.values:
+            row[c.name] = random.choice(c.values)[0]
         else:
             val = fakes[c.data_type]()
             row[c.name] = val
             if c.data_type in {"boro_code", "block", "lot"}:
                 found_bbl_parts[c.data_type] = val
-    if set(found_bbl_parts.keys()) == bbl_parts:
-        row["bbl"] = fakes["bbl"](
-            found_bbl_parts["boro_code"],
-            found_bbl_parts["block"],
-            found_bbl_parts["lot"],
-        )
 
+    # Construct a BBL from found parts, or generate a new one
+    if found_bbl_name:
+        if set(found_bbl_parts.keys()) == bbl_parts:
+            row[found_bbl_name] = fakes["bbl"](
+                found_bbl_parts["boro_code"],
+                found_bbl_parts["block"],
+                found_bbl_parts["lot"],
+            )
+        else:
+            row[found_bbl_name] = fakes["bbl"](
+                fakes["boro_code"](),
+                fakes["block"](),
+                fakes["lot"](),
+            )
+
+    for c in columns:
+        if c.is_nullable and random.choice([True, False]):
+            # adding some extra chaos
+            if random.choice([True, False]):
+                del row[c.name]
+            else:
+                row[c.name] = ""
     return row
 
 
-def _generate_fake_dataset(row_count: int, columns: list[models.Column]):
-    return df.from_records([fake_row(columns) for i in range(row_count)])
+def generate_fake_dataset(row_count: int, columns: list[models.Column]):
+    return df.from_records([_fake_row(columns) for i in range(row_count)])
+
+
+def generate_fake_dataset_from_test_md(row_count: int):
+    dataset = metadata.dataset_package.get_dataset("primary_csv")
+    return generate_fake_dataset(row_count, columns=dataset.get_columns(metadata))
```

Comment:
✂️?

Reply:
Yeah, this is cruft. Added a comment for why I kept it in, mostly to get a fake dataset from test metadata in a Jupyter Notebook without futzing with paths. So just entirely for convenience for the dev process.